wpilibNewCommands/src/test/java/edu/wpi/first/wpilibj2/command/CommandDecoratorTest.java

```diff
@@ -90,61 +154,75 @@ void ignoringDisableTest() {
   @Test
   void beforeStartingTest() {
     try (CommandScheduler scheduler = new CommandScheduler()) {
-      AtomicBoolean condition = new AtomicBoolean();
-      condition.set(false);
+      AtomicBoolean finished = new AtomicBoolean();
+      finished.set(false);
+
+      Command command = new InstantCommand().beforeStarting(() -> finished.set(true));
 
-      Command command = new InstantCommand();
+      scheduler.schedule(command);
 
-      scheduler.schedule(command.beforeStarting(() -> condition.set(true)));
+      assertTrue(finished.get());
 
-      assertTrue(condition.get());
+      scheduler.run();
+
+      assertTrue(scheduler.isScheduled(command));
+
+      scheduler.run();
+
+      assertFalse(scheduler.isScheduled(command));
     }
   }
 
   @Test
   void andThenLambdaTest() {
     try (CommandScheduler scheduler = new CommandScheduler()) {
-      AtomicBoolean condition = new AtomicBoolean();
-      condition.set(false);
+      AtomicBoolean finished = new AtomicBoolean(false);
 
-      Command command = new InstantCommand();
+      Command command = new InstantCommand().andThen(() -> finished.set(true));
 
-      scheduler.schedule(command.andThen(() -> condition.set(true)));
+      scheduler.schedule(command);
 
-      assertFalse(condition.get());
+      assertFalse(finished.get());
 
       scheduler.run();
 
-      assertTrue(condition.get());
+      assertTrue(finished.get());
+
+      scheduler.run();
+
+      assertFalse(scheduler.isScheduled(command));
     }
   }
 
   @Test
   void andThenTest() {
     try (CommandScheduler scheduler = new CommandScheduler()) {
-      AtomicBoolean condition = new AtomicBoolean();
-      condition.set(false);
+      AtomicBoolean condition = new AtomicBoolean(false);
 
       Command command1 = new InstantCommand();
       Command command2 = new InstantCommand(() -> condition.set(true));
+      Command group = command1.andThen(command2);
 
-      scheduler.schedule(command1.andThen(command2));
+      scheduler.schedule(group);
 
       assertFalse(condition.get());
 
       scheduler.run();
 
       assertTrue(condition.get());
+
+      scheduler.run();
+
+      assertFalse(scheduler.isScheduled(group));
```

Comment:
Can you add similar assertions throughout? To see how the sequence's end aligns with the internal commands.

Reply:
What places seem to be missing? The only ones that don't end in `assertFalse(scheduler.isScheduled(group))` are `ignoringDisableTest()`, `unlessTest()`, `onlyIfTest()`, `finallyDoTest()`, `handleInterruptTest()`, and `withNameTest()` (and the equivalents in C++), which I don't think really need the end conditions to be tested. If you have reasons to think differently, though, I'm opening to adding them. (And if there's something more specific you meant, I'd like to hear that too)